lib/features/rules_filter_creator/presentation/rules_filter_creator_controller.dart

```diff
@@ -145,7 +135,18 @@ class RulesFilterCreatorController extends BaseMailboxController {
     }
   }
 
+  @override
+  void onDone() {
+    viewState.value.fold((failure) => null, (success) {
+      if (success is GetAllMailboxSuccess) {
+        _setUpRuleFilterActions();
```

Comment:
Why can't this method be called in `handleSuccessViewState`?

Reply:
Because it will process the incoming data multiple times. Here we only want to process the last data received.